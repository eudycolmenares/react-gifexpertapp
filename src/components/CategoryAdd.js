import { useState } from "react"
import PropTypes from 'prop-types'

const CategoryAdd = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('')
    const handleInputChange = (e) => {
        setInputValue( e.target.value )
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if ( inputValue.trim().length > 3 ) {
            setCategories( categories => [inputValue, ...categories] )
            setInputValue('')
        }
    }
    // 
    return (
        <form onSubmit={ handleSubmit }>
            {/* Temporal */}
            <p>{ inputValue }</p> 
            <input 
                type="text" 
                value={ inputValue } 
                onChange={ handleInputChange }
            />
        </form>
    )
}
CategoryAdd.propTypes = {
    setCategories: PropTypes.func.isRequired
}
// 
export default CategoryAdd