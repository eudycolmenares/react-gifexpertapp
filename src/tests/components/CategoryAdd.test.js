import { shallow } from 'enzyme'
import '@testing-library/jest-dom'
import CategoryAdd from '../../components/CategoryAdd'
// 
describe('Tests in <CategoryAdd />', () => {
    const setCategories = jest.fn()
    let wrapper
    beforeEach(() => {
        jest.clearAllMocks()
        wrapper = shallow( <CategoryAdd setCategories={ setCategories } />)
    })
    test('Should display correctly!', () => {
        expect( wrapper ).toMatchSnapshot()
    })
    test('Should change the input text!', () => {
        const input = wrapper.find('input')
        const value = 'Hola mundo'
        input.simulate('change', { target: {value: value} })
        expect( wrapper.find('p').text() ).toBe( value )
    })
    test('Should not post the information with the submit!', () => {
        wrapper.find( 'form' ).simulate('submit', { preventDefault(){} })
        expect( setCategories ).not.toHaveBeenCalled()
    })
    test('Should called setCategories() and clear the input text!', () => {
        const value = 'Hola mundo'
        wrapper.find('input').simulate('change', { target: {value: value} })
        wrapper.find( 'form' ).simulate('submit', { preventDefault(){} })
        expect( setCategories ).toHaveBeenCalled()
        expect( setCategories ).toHaveBeenCalledTimes(1)
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) )
        expect( wrapper.find('input').prop('value') ).toBe( '' )
    })
})