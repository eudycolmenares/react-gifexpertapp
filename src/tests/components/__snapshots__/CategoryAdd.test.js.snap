// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <CategoryAdd /> Should display correctly! 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
